### PR TITLE
Prepare the 1.2.0 release

### PR DESCRIPTION
**Pull request by ayys**
This PR is to prepare the 1.2.0 release of wasmer-python. Check the changelog committed in this PR for details.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,12 +4,30 @@ All notable changes to this project will be documented in this file.
...
 * [0.4.1](#041---2020-02-02)
 * [0.3.0](#030---2019-07-16)
 * [0.2.0](#020---2019-04-16)
 
+## [1.2.0] - 2023-04-20
+
+## Added
+
+* Added support for Python 3.11 (#696)
+* Added support for Python 3.10 (#680)
```
Comment (Michael-F-Bryan) [inline CHANGELOG.md:19]: Should these link back to the original issues?

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,12 +4,30 @@ All notable changes to this project will be documented in this file.
...
+* Improvements to the documentation ([e880e25](https://github.com/wasmerio/wasmer-python/commit/e880e25))
+
+## Changed
+
+* Updated to the latest version of Maturin (0.14.17) to fix failing Windows builds (#715)
+
+## Fixed
+* Fixed compatibility issues with Python 3.11 (#696)
+* Fixed compatibility issues with Python 3.10 (#680)
+* Fixed typos in the documentation (#639, #681)
```
Comment (Michael-F-Bryan) [inline CHANGELOG.md:29]: Can you add links for these too?

**Review by Michael-F-Bryan — CHANGES_REQUESTED** [review]
Let's add links to the tickets in the changelog. Other than that, I'm happy to do a release.

**Comment by ayys** [comment]
@Michael-F-Bryan I've added links to all the issues I am pointing at. The pre-release CI is still failing though 😞

**Review by Michael-F-Bryan — APPROVED** [review]

**Pull request merged**